apps/maestro/src/server/routers/sui/utils/utils.ts

```diff
@@ -112,10 +112,18 @@ export const getCoinAddressAndManagerByTokenId = async (input: {
       ?.fields?.value?.fields.registered_coins.fields.id.id;
 
     do {
-      result = await suiClient.getDynamicFields({
-        parentId: registeredCoinsBagId,
-        cursor: cursor,
-      });
+      result = await suiClient
+        .getDynamicFields({
+          parentId: registeredCoinsBagId,
+          cursor: cursor,
+        })
+        .catch((error) => {
+          console.error("Failed to get dynamic fields:", error);
+          return null;
+        });
+      if (!result) {
+        return null;
+      }
```

Comment:
can you can convert the whole do-while block into a utility function? we might want to reuse it to handle paginated results.